components/shared/webrender/display_list.rs

```diff
@@ -6,17 +6,38 @@
 
 use embedder_traits::Cursor;
 use serde::{Deserialize, Serialize};
+use style::values::specified::Overflow;
 use webrender_api::units::{LayoutSize, LayoutVector2D};
 use webrender_api::{Epoch, ExternalScrollId, PipelineId, ScrollLocation, SpatialId};
 
-/// The scroll sensitivity of a scroll node ie whether it can be scrolled due to input event and
-/// script events or only script events.
+/// The scroll sensitivity of a scroll node in a particular axis ie whether it can be scrolled due to
+/// input events and script events or only script events.
 #[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
 pub enum ScrollSensitivity {
     /// This node can be scrolled by input and script events.
     ScriptAndInputEvents,
     /// This node can only be scrolled by script events.
     Script,
+    /// This node cannot be scrolled.
+    None,
+}
+
+/// Convert [Overflow] to [ScrollSensitivity].
+impl From<Overflow> for ScrollSensitivity {
+    fn from(overflow: Overflow) -> Self {
+        match overflow {
+            Overflow::Hidden => ScrollSensitivity::Script,
+            Overflow::Scroll | Overflow::Auto => ScrollSensitivity::ScriptAndInputEvents,
+            _ => ScrollSensitivity::None,
+        }
+    }
+}
+
+/// The [ScrollSensitivity] of particular node in the vertical and horizontal axes.
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
+pub struct AxesScrollSensitivity {
+    pub x: ScrollSensitivity,
+    pub y: ScrollSensitivity,
 }
```

Comment:
Nit: There are 3 cases where `AxesScrollSensitivity` is created using the same sensitivity for both axes. Might be worth doing `impl From<ScrollSensitivity> for AxesScrollSensitivity`. But fine either way.